packages/nextjs/src/api/index.ts

```diff
@@ -1,6 +1,65 @@
-if (!process.env.CLERK_API_KEY) {
-  throw Error('The CLERK_API_KEY environment variable must be set to use imports from @clerk/nextjs/api.');
-}
+import {
+  ClerkMiddleware,
+  ClerkMiddlewareOptions,
+  default as Clerk,
+  RequireAuthProp,
+  WithAuthProp,
+} from '@clerk/clerk-sdk-node';
+import type { NextApiRequest, NextApiResponse } from 'next';
 
-export { default } from '@clerk/clerk-sdk-node';
+export default Clerk;
 export * from '@clerk/clerk-sdk-node';
+
+// https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support
+function runMiddleware(req: NextApiRequest, res: NextApiResponse, fn: ClerkMiddleware) {
+  return new Promise((resolve, reject) => {
+    // @ts-ignore
+    fn(req, res, result => {
+      if (result instanceof Error) {
+        return reject(result);
+      }
+      return resolve(result);
+    });
+  });
+}
+
+type NextApiHandlerWithAuth<T = any> = (
+  req: WithAuthProp<NextApiRequest>,
+  res: NextApiResponse<T>,
+) => void | Promise<void>;
+
+// Set the session on the request and then call provided handler
+export function withAuth(handler: NextApiHandlerWithAuth, options?: ClerkMiddlewareOptions): NextApiHandlerWithAuth {
```

Comment:
🙃 
We can also put `withAuth` and `requireAuth` in a separate file and keep the index just for imports/exports. WDYT ?

Reply:
Good idea. Will do.